### PR TITLE
DD-796 update service dependencies

### PR DESCRIPTION
**Pull request by rcraig14**
There are a lot of changes in this one, sorry in advance! 

Notable changes:
- Update Node version to match serverless runtime
  - erbium (node12)  -> hydrogen (node 18) in .nvmrc
  - nodejs16 -> nodejs18in serverless.ts
- Removed dependency to  therockstorm/utils. 
  - Updated log calls to use `console.log`, `console.error`, `console.warn` since this is what the library was doing anyways.
  - Updated  `envVar` to new function [envVarRequired](https://github.com/Dwolla/webhook-provisioner/blob/3146845d35b2cd9a4a72d3b37988631a9060b4e6/src/util.ts#L6-L12) in utils
- Removed dependency to skripts
  - Update project to have linting capabilities by calling eslint directly
  - Update project to have formatting capabilities by calling prettier directly
  - Update project precommit hooks to call lint staged directly
  - Update serverless.ts to 
  - Add and update dependencies required that we no longer get for "free" with this library
  - Update serverless file to include [default values](https://github.com/therockstorm/skripts/blob/master/src/config/serverless.js) not being overridden
  - Update webpack to include [default values](https://github.com/therockstorm/skripts/blob/master/src/config/webpack.config.js) not being overridden
  - Update prettier config to include [default values](https://github.com/therockstorm/skripts/blob/master/src/config/prettier.config.js) not being overridden
  - Update lintstaged config to include [default values](https://github.com/therockstorm/skripts/blob/master/src/config/lintstagedrc.js) not being overridden
  - Add and update linter config to include [default values](https://github.com/therockstorm/skripts/blob/master/src/config/lintstagedrc.js) not being overridden
- Removed serverless plugin `serverless-pseudo-parameters`
  - Followed [documentation from plugin docs](https://www.npmjs.com/package/serverless-pseudo-parameters) for handling deprecated code. This can be seen in the serverless file
  ```
  With version v2.50.0, new variables sources ${aws:accountId} and ${aws:region} were introduced, which can be used in properties where CloudFormation pseudo paramaters cannot be used. Please use them instead of #{AWS::...} format as supported by this plugin
  ```
- Linter ran and made a lot of updates to add `;`

### CODE REVIEW

File: .nvmrc
Comment (lshepherd-dwolla) [inline .nvmrc:1]: anything stopping us from going v20? Would give us a [long maintenance runway](https://nodejs.dev/en/about/releases/)

Comment (rcraig14) [reply .nvmrc:1]: Right now v20 is not supported [by aws](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html)

**Review by lshepherd-dwolla — APPROVED** [review]
LGTM based on changes here - not sure if you had anything still in the works

**Comment by rcraig14** [comment]
@lshepherd-dwolla I added one more commit. A few things in the serverless file were deprecated and just needed simple updates.

**Review by lmshepherd — APPROVED** [review]

**Review by lmshepherd — DISMISSED** [review]

**Comment by rcraig14** [comment]
Closing and reopening PR to fix build issue

**Pull request closed without merging**